staging/src/k8s.io/legacy-cloud-providers/aws/aws.go

```diff
@@ -1425,10 +1425,17 @@ func (c *Cloud) NodeAddresses(ctx context.Context, name types.NodeName) ([]v1.No
 				continue
 			}
 			ipPath := path.Join("network/interfaces/macs/", macID, "local-ipv4s")
-			macIPs[num], err = c.metadata.GetMetadata(ipPath)
+			v4ips, err := c.metadata.GetMetadata(ipPath)
 			if err != nil {
 				return nil, fmt.Errorf("error querying AWS metadata for %q: %q", ipPath, err)
 			}
+			ipPath = path.Join("network/interfaces/macs/", macID, "ipv6s")
+			v6ips, err := c.metadata.GetMetadata(ipPath)
+			if err != nil {
+				klog.V(3).Infof("Error querying IPv6 IPs: %v", err)
+				v6ips = ""
+			}
+			macIPs[num] = v4ips + "\n" + v6ips
```

Comment:
So this would mean that if `eth0` was IPv6 only, and `eth1` is dual-stack, then with the old code, `node.Status.Addresses[0].Address` would be the `eth1` IPv4 IP, but with the new code it would be the `eth0` IPv6 IP.

I'm not sure if that's actually possible though (IPv6-only `eth0`)?

Assuming `eth0` has an IPv4 address, then the new code might change the total overall ordering of node addresses, but shouldn't change the "primary" address, so it hopefully shouldn't matter.

Reply:
I think it is not possible as today, AWS is always dual-stack with ipv6, but better to confirm

This is the output of one my instance in aws with IPv6
```
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 9001 qdisc pfifo_fast state UP group default qlen 1000
    link/ether 06:bf:36:fe:92:a2 brd ff:ff:ff:ff:ff:ff
    inet 10.0.1.173/24 brd 10.0.1.255 scope global eth0
       valid_lft forever preferred_lft forever
    inet6 2a05:d012:db0:3f01:4f2a:bfc1:96ac:db2a/128 scope global
       valid_lft forever preferred_lft forever
    inet6 fe80::4bf:36ff:fefe:92a2/64 scope link
       valid_lft forever preferred_lft forever
```

Reply:
According to the AWS [docs](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-instance-addressing.html) IPv4 is required:

> By default, Amazon EC2 and Amazon VPC use the IPv4 addressing protocol; you can't disable this behavior. When you create a VPC, you must specify an IPv4 CIDR block (a range of private IPv4 addresses).